cmd/metal-api/main.go

```diff
@@ -907,8 +910,25 @@ func evaluateVPNConnected() error {
 		if m.Allocation == nil || m.Allocation.VPN == nil {
 			continue
 		}
-		connected := connectedMap[m.ID]
+
+		connected := slices.ContainsFunc(headscaleMachines, func(headscaleMachine *headscalev1.Machine) bool {
```

Comment:
It would be better if we can keep the headscale dependency in the headscale package.

Reply:
I wanted to move this to the service layer similar to the other evaluation commands and in order to be able to write a test. Maybe now it's better.